db/log_writer.cc

```diff
@@ -89,8 +93,15 @@ Status Writer::EmitPhysicalRecord(RecordType t, const char* ptr, size_t n) {
 
   // Compute the crc of the record type and the payload.
   uint32_t crc = crc32c::Extend(type_crc_[t], ptr, n);
-  crc = crc32c::Mask(crc);                 // Adjust for storage
-  EncodeFixed32(buf, crc);
+  if (db_options_->recycle_log_files) {
```

Comment:
If you feel it should be the way to go, then leave it. Not big deal. I still feel if we the giant object DBOptions, it is harder for us to reason about the module dependencies.